### PR TITLE
Update to latest logging format and modify the accuracy method.

### PR DESCRIPTION
**Pull request by limberc**
## Update to latest logging format and modify the accuracy method.

Update to latest logging format in the ImageNet example and modify the accuracy method.

The accuracy method has been modified in https://github.com/pytorch/examples/blob/master/imagenet/main.py#L423 which they use reshape instead of view. I have found that in pytorch 1.7.0, setting the original line, it would get an error.

Fixes # (issue)
1. logging method has been replaced to latest self.log format.
2. replace `view` to `reshape`.

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: pl_examples/domain_templates/imagenet.py [hunk 1]
```diff
@@ -76,37 +74,19 @@ def training_step(self, batch, batch_idx):
         output = self(images)
         loss_val = F.cross_entropy(output, target)
```
Comment (Borda) [inline pl_examples/domain_templates/imagenet.py:75]:
```suggestion
        loss_train= F.cross_entropy(output, target)
```

File: pl_examples/domain_templates/imagenet.py [hunk 2]
```diff
@@ -76,37 +74,19 @@ def training_step(self, batch, batch_idx):
...
-        tqdm_dict = {'train_loss': loss_val}
-        output = OrderedDict({
-            'loss': loss_val,
-            'acc1': acc1,
-            'acc5': acc5,
-            'progress_bar': tqdm_dict,
-            'log': tqdm_dict
-        })
-        return output
+        self.log('train_loss', loss_val, on_step=True, on_epoch=True, logger=True)
```
Comment (Borda) [inline pl_examples/domain_templates/imagenet.py:77]:
```suggestion
        self.log('train_loss', loss_train, on_step=True, on_epoch=True, logger=True)
```

**Review by Borda — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

File: pl_examples/domain_templates/imagenet.py [hunk 3]
```diff
@@ -121,7 +101,7 @@ def __accuracy(output, target, topk=(1,)):
 
             res = []
             for k in topk:
-                correct_k = correct[:k].view(-1).float().sum(0, keepdim=True)
+                correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
```
Comment (justusschock) [inline pl_examples/domain_templates/imagenet.py:104]: A general question: Shouldn't we rather use the functional interface here since we already got this? @Borda @tchaton 

If we want to keep it like that, that's also fine with me :)

**Review by justusschock — APPROVED** [review]

File: pl_examples/domain_templates/imagenet.py (hunk 3 above)
Comment (limberc) [reply pl_examples/domain_templates/imagenet.py:104]: Well, I think it looks great for an ImageNet beginner since the accuracy method is as the same as the official one.

**Review by SkafteNicki — APPROVED** [review]

File: pl_examples/domain_templates/imagenet.py [hunk 4]
```diff
@@ -74,39 +72,21 @@ def forward(self, x):
     def training_step(self, batch, batch_idx):
         images, target = batch
         output = self(images)
-        loss_val = F.cross_entropy(output, target)
+        loss_train= F.cross_entropy(output, target)
```
Comment (awaelchli) [inline pl_examples/domain_templates/imagenet.py:75]:
```suggestion
        loss_train = F.cross_entropy(output, target)
```
whitespace, check the code formatter for remaining issue with PEP

**Review by awaelchli — APPROVED** [review]
